src/arrayMethodJoin.js

```diff
@@ -4,8 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let str = '';
+
+    for (let i = 0; i < this.length - 1; i++) {
+      if (this[i] !== null && this[i] !== undefined) {
+        str += this[i];
+      }
+
+      if (i !== this.length - 1) {
+        str += separator;
+      }
+    }
+
+    if (this[this.length - 1] !== undefined) {
+      str += this[this.length - 1];
+    }
```

Comment:
why do you need this condition? It looks redundant

Reply:
The code doesn't work without the lines 20-22

Reply:
for the future: if you don't agree or have questions, write in the chat. it will be faster